test/js/node/http/node-http.test.ts

```diff
@@ -823,4 +826,41 @@ describe("node:http", () => {
       }
     });
   });
+
+  test("dns lookup localhost, issue#4358", done => {
+    const { mustCall, mustNotCall } = createCallCheckCtx(done);
+
+    const host = "localhost";
+
+    const server = createServer();
+
+    let timeout: Timer;
+    const closeAndFail = (message?: string) => {
+      clearTimeout(timeout);
+      server.close();
+      mustNotCall(message)();
+    };
+    server.on("error", closeAndFail);
+    timeout = setTimeout(closeAndFail, 100);
+
+    dns.lookup(host, (err, ip, family) => {
+      if (err) {
+        return closeAndFail("dns lookup failed");
+      }
+
+      server.listen(
+        0,
+        host,
+        mustCall(() => {
+          const address = server.address() as AddressInfo;
+          expect(address.address).toStrictEqual(ip);
```

Comment:
this test is failing

Reply:
https://github.com/oven-sh/bun/actions/runs/6002838901/job/16325640437#step:5:6184